### PR TITLE
Cppcheck minor updates.

### PR DESCRIPTION
**Pull request by plbossart**
cppcheck --platform=unix64 --force --max-configs=1024 --inconclusive --enable=all --suppress=variableScope --suppress=shiftTooManyBitsSigned --suppress=arithOperationsOnVoidPointer --suppress=bitwiseOnBoolean sound/soc/sof/

### CODE REVIEW

File: sound/soc/sof/sof-audio.c [hunk 1]
```diff
@@ -881,20 +881,6 @@ struct snd_sof_pcm *snd_sof_find_spcm_comp(struct snd_soc_component *scomp,
 	return NULL;
 }
 
-struct snd_sof_pcm *snd_sof_find_spcm_pcm_id(struct snd_soc_component *scomp,
-					     unsigned int pcm_id)
```
Comment (ujfalusi) [inline sound/soc/sof/sof-audio.c:856]: remove it from `sof-audio.h` as well?

**Review by ujfalusi — CHANGES_REQUESTED** [review]
@plbossart, looks ok, but the header should be cleaned as well.

File: sound/soc/sof/sof-audio.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/sof/sof-audio.c:856]: yes, it's not static as I mistakenly thought. fixed now.

**Review by ujfalusi — APPROVED** [review]

**Comment by plbossart** [comment]
rebased to solve conflict with Ranjani's earlier cleanup.

**Review by ranj063 — APPROVED** [review]

**Comment by plbossart** [comment]
@ujfalusi @ajitkupandey can you take a look? Thanks!

**Review by ujfalusi — APPROVED** [review]
@plbossart, sorry for the delay, I thought that I have added my updated review after the you have pushed it.

**Pull request merged**